<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from hexbot.xacro                   | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="hexbot">
  <link name="base_link">
    <inertial>
      <origin rpy="0 0 0" xyz="-5.7578E-17 -0.026865 -5.6409E-17"/>
      <mass value="0.45569"/>
      <inertia ixx="0.00067888" ixy="-1.6518E-19" ixz="4.3182E-15" iyy="0.0012021" iyz="-3.5752E-19" izz="0.00067888"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hexbot/meshes/base_link.STL"/>
      </geometry>
      <material name="">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hexbot/meshes/base_link.STL"/>
      </geometry>
    </collision>
  </link>
  <link name="thigh_1">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.0416454946793614"/>
      <inertia ixx="1.01464532907881E-05" ixy="-9.01102352886516E-08" ixz="5.83026179577091E-08" iyy="9.79635939418219E-06" iyz="2.29962924939912E-11" izz="1.11686383627586E-05"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hexbot/meshes/thigh.STL"/>
      </geometry>
      <material name="">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hexbot/meshes/thigh.STL"/>
      </geometry>
    </collision>
  </link>
  <link name="thigh_2">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.0416454946793614"/>
      <inertia ixx="1.01464532907881E-05" ixy="-9.01102352886516E-08" ixz="5.83026179577091E-08" iyy="9.79635939418219E-06" iyz="2.29962924939912E-11" izz="1.11686383627586E-05"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hexbot/meshes/thigh.STL"/>
      </geometry>
      <material name="">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hexbot/meshes/thigh.STL"/>
      </geometry>
    </collision>
  </link>
  <link name="thigh_3">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.0416454946793614"/>
      <inertia ixx="1.01464532907881E-05" ixy="-9.01102352886516E-08" ixz="5.83026179577091E-08" iyy="9.79635939418219E-06" iyz="2.29962924939912E-11" izz="1.11686383627586E-05"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hexbot/meshes/thigh.STL"/>
      </geometry>
      <material name="">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hexbot/meshes/thigh.STL"/>
      </geometry>
    </collision>
  </link>
  <link name="thigh_4">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.0416454946793614"/>
      <inertia ixx="1.01464532907881E-05" ixy="-9.01102352886516E-08" ixz="5.83026179577091E-08" iyy="9.79635939418219E-06" iyz="2.29962924939912E-11" izz="1.11686383627586E-05"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hexbot/meshes/thigh.STL"/>
      </geometry>
      <material name="">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hexbot/meshes/thigh.STL"/>
      </geometry>
    </collision>
  </link>
  <link name="thigh_5">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.0416454946793614"/>
      <inertia ixx="1.01464532907881E-05" ixy="-9.01102352886516E-08" ixz="5.83026179577091E-08" iyy="9.79635939418219E-06" iyz="2.29962924939912E-11" izz="1.11686383627586E-05"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hexbot/meshes/thigh.STL"/>
      </geometry>
      <material name="">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hexbot/meshes/thigh.STL"/>
      </geometry>
    </collision>
  </link>
  <link name="thigh_6">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.0416454946793614"/>
      <inertia ixx="1.01464532907881E-05" ixy="-9.01102352886516E-08" ixz="5.83026179577091E-08" iyy="9.79635939418219E-06" iyz="2.29962924939912E-11" izz="1.11686383627586E-05"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hexbot/meshes/thigh.STL"/>
      </geometry>
      <material name="">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hexbot/meshes/thigh.STL"/>
      </geometry>
    </collision>
  </link>
  <link name="shin_1">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.0416454939081647"/>
      <inertia ixx="1.01464529463005E-05" ixy="-5.79008683658633E-11" ixz="1.48086764022737E-07" iyy="1.42061146123856E-05" iyz="-6.47788995139536E-12" izz="6.7588826935067E-06"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hexbot/meshes/shin.STL"/>
      </geometry>
      <material name="">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hexbot/meshes/shin.STL"/>
      </geometry>
    </collision>
  </link>
  <link name="shin_2">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.0416454939081647"/>
      <inertia ixx="1.01464529463005E-05" ixy="-5.79008683658633E-11" ixz="1.48086764022737E-07" iyy="1.42061146123856E-05" iyz="-6.47788995139536E-12" izz="6.7588826935067E-06"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hexbot/meshes/shin.STL"/>
      </geometry>
      <material name="">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hexbot/meshes/shin.STL"/>
      </geometry>
    </collision>
  </link>
  <link name="shin_3">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.0416454939081647"/>
      <inertia ixx="1.01464529463005E-05" ixy="-5.79008683658633E-11" ixz="1.48086764022737E-07" iyy="1.42061146123856E-05" iyz="-6.47788995139536E-12" izz="6.7588826935067E-06"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hexbot/meshes/shin.STL"/>
      </geometry>
      <material name="">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hexbot/meshes/shin.STL"/>
      </geometry>
    </collision>
  </link>
  <link name="shin_4">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.0416454939081647"/>
      <inertia ixx="1.01464529463005E-05" ixy="-5.79008683658633E-11" ixz="1.48086764022737E-07" iyy="1.42061146123856E-05" iyz="-6.47788995139536E-12" izz="6.7588826935067E-06"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hexbot/meshes/shin.STL"/>
      </geometry>
      <material name="">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hexbot/meshes/shin.STL"/>
      </geometry>
    </collision>
  </link>
  <link name="shin_5">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.0416454939081647"/>
      <inertia ixx="1.01464529463005E-05" ixy="-5.79008683658633E-11" ixz="1.48086764022737E-07" iyy="1.42061146123856E-05" iyz="-6.47788995139536E-12" izz="6.7588826935067E-06"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hexbot/meshes/shin.STL"/>
      </geometry>
      <material name="">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hexbot/meshes/shin.STL"/>
      </geometry>
    </collision>
  </link>
  <link name="shin_6">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.0416454939081647"/>
      <inertia ixx="1.01464529463005E-05" ixy="-5.79008683658633E-11" ixz="1.48086764022737E-07" iyy="1.42061146123856E-05" iyz="-6.47788995139536E-12" izz="6.7588826935067E-06"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hexbot/meshes/shin.STL"/>
      </geometry>
      <material name="">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hexbot/meshes/shin.STL"/>
      </geometry>
    </collision>
  </link>
  <link name="foot_1">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.0264807627517247"/>
      <inertia ixx="5.16912819060422E-06" ixy="-1.91682876233419E-10" ixz="5.8167586632756E-08" iyy="7.99193660112382E-06" iyz="7.92351583964294E-12" izz="4.71320299684973E-06"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hexbot/meshes/foot.STL"/>
      </geometry>
      <material name="">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hexbot/meshes/foot.STL"/>
      </geometry>
    </collision>
  </link>
  <link name="foot_2">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.0264807627517247"/>
      <inertia ixx="5.16912819060422E-06" ixy="-1.91682876233419E-10" ixz="5.8167586632756E-08" iyy="7.99193660112382E-06" iyz="7.92351583964294E-12" izz="4.71320299684973E-06"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hexbot/meshes/foot.STL"/>
      </geometry>
      <material name="">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hexbot/meshes/foot.STL"/>
      </geometry>
    </collision>
  </link>
  <link name="foot_3">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.0264807627517247"/>
      <inertia ixx="5.16912819060422E-06" ixy="-1.91682876233419E-10" ixz="5.8167586632756E-08" iyy="7.99193660112382E-06" iyz="7.92351583964294E-12" izz="4.71320299684973E-06"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hexbot/meshes/foot.STL"/>
      </geometry>
      <material name="">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hexbot/meshes/foot.STL"/>
      </geometry>
    </collision>
  </link>
  <link name="foot_4">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.0264807627517247"/>
      <inertia ixx="5.16912819060422E-06" ixy="-1.91682876233419E-10" ixz="5.8167586632756E-08" iyy="7.99193660112382E-06" iyz="7.92351583964294E-12" izz="4.71320299684973E-06"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hexbot/meshes/foot.STL"/>
      </geometry>
      <material name="">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hexbot/meshes/foot.STL"/>
      </geometry>
    </collision>
  </link>
  <link name="foot_5">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.0264807627517247"/>
      <inertia ixx="5.16912819060422E-06" ixy="-1.91682876233419E-10" ixz="5.8167586632756E-08" iyy="7.99193660112382E-06" iyz="7.92351583964294E-12" izz="4.71320299684973E-06"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hexbot/meshes/foot.STL"/>
      </geometry>
      <material name="">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hexbot/meshes/foot.STL"/>
      </geometry>
    </collision>
  </link>
  <link name="foot_6">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.0264807627517247"/>
      <inertia ixx="5.16912819060422E-06" ixy="-1.91682876233419E-10" ixz="5.8167586632756E-08" iyy="7.99193660112382E-06" iyz="7.92351583964294E-12" izz="4.71320299684973E-06"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hexbot/meshes/foot.STL"/>
      </geometry>
      <material name="">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hexbot/meshes/foot.STL"/>
      </geometry>
    </collision>
  </link>
  <joint name="thigh_1_joint" type="revolute">
    <origin rpy="-1.5708 -1.0472 0" xyz="0.0557 -0.049 0.096475"/>
    <parent link="base_link"/>
    <child link="thigh_1"/>
    <axis xyz="0 0 1"/>
    <limit effort="0" lower="-1" upper="1" velocity="0"/>
  </joint>
  <joint name="thigh_2_joint" type="revolute">
    <origin rpy="-1.5707963267949 0 0" xyz="0.111400000000001 -0.051 0"/>
    <parent link="base_link"/>
    <child link="thigh_2"/>
    <axis xyz="0 0 1"/>
    <limit effort="0" lower="-1" upper="1" velocity="0"/>
  </joint>
  <joint name="thigh_3_joint" type="revolute">
    <origin rpy="-1.5707963267949 0.966273587280999 0" xyz="0.0556999999999999 -0.051 -0.0964752299815872"/>
    <parent link="base_link"/>
    <child link="thigh_3"/>
    <axis xyz="0 0 1"/>
    <limit effort="0" lower="-1" upper="1" velocity="0"/>
  </joint>
  <joint name="thigh_4_joint" type="revolute">
    <origin rpy="1.5708 1.0472 -3.1416" xyz="-0.0557 -0.051 -0.096475"/>
    <parent link="base_link"/>
    <child link="thigh_4"/>
    <axis xyz="0 0 1"/>
    <limit effort="0" lower="-1" upper="1" velocity="0"/>
  </joint>
  <joint name="thigh_5_joint" type="revolute">
    <origin rpy="1.5708 0 3.1416" xyz="-0.1114 -0.051 0"/>
    <parent link="base_link"/>
    <child link="thigh_5"/>
    <axis xyz="0 0 1"/>
    <limit effort="0" lower="-1" upper="1" velocity="0"/>
  </joint>
  <joint name="thigh_6_joint" type="revolute">
    <origin rpy="1.5708 -1.0472 3.1416" xyz="-0.0557 -0.051 0.096475"/>
    <parent link="base_link"/>
    <child link="thigh_6"/>
    <axis xyz="0 0 1"/>
    <limit effort="0" lower="-1" upper="1" velocity="0"/>
  </joint>
  <joint name="shin_1_joint" type="revolute">
    <origin rpy="1.5707963267949 -0.785398163397449 0" xyz="0.0648 0.02575 0.02475"/>
    <parent link="thigh_1"/>
    <child link="shin_1"/>
    <axis xyz="0 0 1"/>
    <limit effort="0" lower="-1" upper="1" velocity="0"/>
  </joint>
  <joint name="shin_2_joint" type="revolute">
    <origin rpy="1.5708 -0.7854 0" xyz="0.0648 0.02575 0.02675"/>
    <parent link="thigh_2"/>
    <child link="shin_2"/>
    <axis xyz="0 0 1"/>
    <limit effort="0" lower="-1" upper="1" velocity="0"/>
  </joint>
  <joint name="shin_3_joint" type="revolute">
    <origin rpy="1.5708 -0.7854 0" xyz="0.0648 0.02575 0.02675"/>
    <parent link="thigh_3"/>
    <child link="shin_3"/>
    <axis xyz="0 0 1"/>
    <limit effort="0" lower="-1" upper="1" velocity="0"/>
  </joint>
  <joint name="shin_4_joint" type="revolute">
    <origin rpy="1.5707963267949 -0.785398163397448 0" xyz="0.0647999999999991 0.02575 0.02675"/>
    <parent link="thigh_4"/>
    <child link="shin_4"/>
    <axis xyz="0 0 1"/>
    <limit effort="0" lower="-1" upper="1" velocity="0"/>
  </joint>
  <joint name="shin_5_joint" type="revolute">
    <origin rpy="1.5708 -0.7854 0" xyz="0.0648 0.02575 0.02675"/>
    <parent link="thigh_5"/>
    <child link="shin_5"/>
    <axis xyz="0 0 1"/>
    <limit effort="0" lower="-1" upper="1" velocity="0"/>
  </joint>
  <joint name="shin_6_joint" type="revolute">
    <origin rpy="1.5708 -0.7854 0" xyz="0.0648 0.02575 0.02675"/>
    <parent link="thigh_6"/>
    <child link="shin_6"/>
    <axis xyz="0 0 1"/>
    <limit effort="0" lower="-1" upper="1" velocity="0"/>
  </joint>
  <joint name="foot_1_joint" type="revolute">
    <origin rpy="0 0 0.61087" xyz="0.0647 -0.001 0"/>
    <parent link="shin_1"/>
    <child link="foot_1"/>
    <axis xyz="0 0 1"/>
    <limit effort="0" lower="-1" upper="1" velocity="0"/>
  </joint>
  <joint name="foot_2_joint" type="revolute">
    <origin rpy="0 0 0.610865238198015" xyz="0.0646999999999998 -0.000999999999999224 0"/>
    <parent link="shin_2"/>
    <child link="foot_2"/>
    <axis xyz="0 0 1"/>
    <limit effort="0" lower="-1" upper="1" velocity="0"/>
  </joint>
  <joint name="foot_3_joint" type="revolute">
    <origin rpy="0 0 0.610865238198017" xyz="0.0647000000000003 -0.00099999999999964 0"/>
    <parent link="shin_3"/>
    <child link="foot_3"/>
    <axis xyz="0 0 1"/>
    <limit effort="0" lower="-1" upper="1" velocity="0"/>
  </joint>
  <joint name="foot_4_joint" type="revolute">
    <origin rpy="0 0 0.61087" xyz="0.0647 -0.001 0"/>
    <parent link="shin_4"/>
    <child link="foot_4"/>
    <axis xyz="0 0 1"/>
    <limit effort="0" lower="-1" upper="1" velocity="0"/>
  </joint>
  <joint name="foot_5_joint" type="revolute">
    <origin rpy="0 0 0.610865238198015" xyz="0.0647000000000012 -0.00100000000000069 0"/>
    <parent link="shin_5"/>
    <child link="foot_5"/>
    <axis xyz="0 0 1"/>
    <limit effort="0" lower="-1" upper="1" velocity="0"/>
  </joint>
  <joint name="foot_6_joint" type="revolute">
    <origin rpy="0 0 0.61087" xyz="0.0647 -0.001 0"/>
    <parent link="shin_6"/>
    <child link="foot_6"/>
    <axis xyz="0 0 1"/>
    <limit effort="0" lower="-1" upper="1" velocity="0"/>
  </joint>
  <link name="base_footprint">
  </link>
  <joint name="base_joint" type="fixed">
    <origin rpy="-1.5708 0 0" xyz="0 0 0.25"/>
    <parent link="base_footprint"/>
    <child link="base_link"/>
  </joint>
</robot>

